motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -463,24 +436,26 @@ int Ros_MotionServer_SimpleMsgProcess(Controller* controller, SimpleMsg* receive
 		ret = Ros_MotionServer_JointTrajPtFullExProcess(controller, receiveMsg, replyMsg);
 		break;
 
+
+//Maintain backward compatibility for users who are sending I/O over motion-server

```

Comment:
@gavanderhoorn  Here is where I continue to process I/O requests on the motion server.  It just calls the io function from ioserver.c

Reply:
O, that is very nice. I'd missed that. :+1: 

Are the MotoPlus IO functions thread safe? If `MotionServer` and `IoServer` are run in different threads, would `*_ReadIOBit(..)` (and the others) need to introduce locking/mutexes?